alfasim_sdk_api/api.h

```diff
@@ -592,4 +592,69 @@ DLL_EXPORT int get_wall_layer_id(void* ctx, int control_volume, const char* mate
 */
 DLL_EXPORT int set_wall_layer_property(void* ctx, int control_volume, int wall_layer_id, enum WallLayerProperty property_id, double new_value);
 
+/*!
+    Gets the current UCM (unit cell model) input data for friction factor calculation.
+    Any available variable by this function is considered for a unit cell, which means
+    that there are variables with one value and there are varailbes with two values
+    related to the two phase system (GAS and LIQUID).
+    If a given variable name is not available, a #NOT_AVAILABLE_DATA error is returned.
+
+    List of `variable_name` with two values (Two phase):
+    - `"alpha"`: Volume Fraction [m3 of `phase` /m3 of mixture]
+    - `"rho"`: Density [kg/m3]
+    - `"mu"`: Viscosity [Pa.s]
+    - `"U"`: Velocity [m/s]
+
+    It is important to know that the listed `variable_name`s are not available in any phase, only for
+    two phase systems, in which `Gas id` is 0 (zero) and `Liquid id` (sum of all liquid phases) is 1 (One).
+    Because of that, the #error_code must be checked.
+
+    List of `variable_name` with one value:
+    - `"D"`: Unit Cell Inner Diameter [m]
+    - `"ks"`: Roughness [m]
+    - `"theta"`: Inclination of the Unit Cell [rad]
+    - `"sigma"`: Gas-liquid Surface Tension [N/m]
+
+    @param[in] ctx ALFAsim's plugins context.
+    @param[out] out Variable values array.
+    @param[in] var_name String with the variable name. See the list of possible values above.
+    @param[in] phase_id Phase Id, When it is Gas phase the Id must be 0
+                          when it is Liquid phase the Id must be 1.
+    @return An #error_code value.
+*/
+DLL_EXPORT int get_ucm_friction_factor_input_variable(
+    void* ctx, double* out, const char* var_name, int phase_id
+);
+
+/*!
+    Gets the current UCM (unit cell model) fluid geometrical properties for friction
+    factor calculation.
+
+    During the implementation of any HOOK related to the UCM friction factor, this
+    function provides the following fluid geometrical properties:
+    - `"S_w"`: Wetted perimeters of phases [m].
+    - `"S_i"`: Interface perimeter [m].
+    - `"H"`: Phase height [m].
+
+    @param[in] ctx ALFAsim's plugins context.
+    @param[out] S_w Wetted Perimeters [m].
+    @param[out] S_i Interface Perimeter [m].
+    @param[out] H   Phase height [m]. For annular flow, H[0] is the core diameter and H[1] is two
```

Comment:
Is it common to be "two times the liquid film"? Could we divide by two and pass the liquid film instead? Does it make any sense?

Reply:
I think it is not the usual way to deal with annular flows in Unit Cell Model. 
Do you agree @arthursoprana ?

Reply:
When I first read the comment I was thinking that the ideal would be `H` to be the liquid film. But then the definitions would change for stratified and annular geometries. In the literature the liquid film thickness is usually represented by `δ`. I currently don't know what would be better and probably if I implement any friction factor model in the future I will always need to check the definition of `H` since it is not the film thickness anyway.

If we change, for the annular geometry `H` to be the actual film thickness, we'll need to review the friction factor equations in order to make sure they  are not already multiplying by `0.5`.

If you don't want to change that, perhaps call it `Total phase height`.

Reply:
- [ ] `Total phase height`